libraries/rush-lib/src/api/CustomTipsConfiguration.ts

```diff
@@ -0,0 +1,83 @@
+import { ConsoleTerminalProvider, ITerminal, JsonFile, Terminal } from '@rushstack/node-core-library';
+
+export interface IRushCustomTipsJson {
+  prefix?: string;
+  customTips?: ICustomTipItemJson[];
+}
+
+/** Similar to {@link TerminalProviderSeverity} but without "debug" and "verbose" */
+export enum TipSeverity {
+  log = 0,
+  warning = 1,
+  error = 2
+}
+
+export interface ICustomTipItemJson {
+  id: CustomTipId;
+  tip: string;
+  prefix?: string;
+  severity?: TipSeverity;
+}
+
+// TODO: consider making this work with the plugin (e.g., the plugins are able to define their own customizable tips)
+export type CustomTipId = 'PNPM_MISMATCH_DEPENDENCY' | 'ANOTHER_ERROR_ID';
+
+export class CustomTipsConfiguration {
+  private _config: IRushCustomTipsJson;
+  private _tipMap: Map<CustomTipId, ICustomTipItemJson>;
+  private _terminal: ITerminal;
+
+  public constructor(configFilename: string) {
+    this._tipMap = new Map();
+    this._config = {};
+    this._terminal = new Terminal(new ConsoleTerminalProvider());
+    this._loadConfig(configFilename);
+  }
+
+  private _loadConfig(configFilename: string): void {
+    try {
+      this._config = JsonFile.load(configFilename);
+    } catch (e) {
+      return;
+    }
+
+    if (!this._config || !this._config.customTips || this._config.customTips?.length === 0) {
```

Comment:
We usually define a schema and validate it via the JsonSchema class. Defining the schema also helps with authoring, since many editors support loading them.

Reply:
Done. May I know how do I get a link like this: `https://developer.microsoft.com/json-schemas/rush/v5/rush-custom-tips.schema.json`?

Reply:
I think the process is somewhat manual but it starts by defining the schema in the repository alongside the existing schemas. Once we have published a version of rush with the schema it then goes through a process to end up with a public URL.